From a9f1138e9068c3b30ef87657f22f8fefe3ff33c6 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 26 Feb 2018 17:00:48 +0200
Subject: [PATCH 0583/1051] dts: ap810: add reserved memory for CM3 firmware

The 0x8000000 --> 0x8100000 memory range is used by the
CM3 firmware for Marvell IPs maintenance, so it should
be reserved and not used by Linux Kernel.

Change-Id: I97f97ccc3722f0f2f791260cc32664c35ec48566
Signed-off-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi b/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi
index db7a65840856..00618cc1bf2d 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi
@@ -29,6 +29,11 @@
 			reg = <0 0x4000000 0 0x200000>;
 			no-map;
 		};
+
+		maintenance-area@8000000 {
+			reg = <0 0x8000000 0 0x100000>;
+			no-map;
+		};
 	};
 
 	config-space {
-- 
2.17.1

